rir/src/runtime/DispatchTable.h

```diff
@@ -213,17 +214,37 @@ struct DispatchTable
         return userDefinedContext_ | anotherContext;
     }
 
+    void blacklistContext(unsigned long con) {
+        if (blacklistIndex_ < BLACKLIST_SIZE) {
+            contextBlacklist_[blacklistIndex_++] = con;
+        }
+    }
+
+    bool isBlacklisted(const Context & c) {
```

Comment:
you could keep the list sorted, then you can abort the search once your entry is larger than the key?